napari/layers/transforms.py

```diff
@@ -0,0 +1,107 @@
+        return self.func(coords)
+
+    def set_slice(self, axes: Sequence[int]):
+        """Return a transform subset to the visible dimensions."""
+        raise NotImplementedError('Cannot subset arbitrary transforms.')
+
+    def set_pad(self, axes: Sequence[int]):
+        """Return a transform with added axes for non-visible dimensions."""
+        raise NotImplementedError('Cannot subset arbitrary transforms.')
+
+    @property
+    def inverse(self):
+        if self._inverse_func is not None:
+            return Transform(self._inverse_func, self.func)
+        else:
+            raise ValueError('Inverse function was not provided.')
+
+
+class ScaleTranslate(Transform):
+    """n-dimensional scale and translation (shift) class.
+
+    Scaling is always applied before translation.
+
+    Parameters
+    ----------
+    scale : 1-D array
+        A 1-D array of factors to scale each axis by. Scale is broadcast to 1
+        in leading dimensions, so that, for example, a scale of [4, 18, 34] in
+        3D can be used as a scale of [1, 4, 18, 34] in 4D without modification.
+        An empty translation vector implies no scaling.
+    translate : 1-D array
+        A 1-D array of factors to shift each axis by. Translation is broadcast
+        to 0 in leading dimensions, so that, for example, a translation of
+        [4, 18, 34] in 3D can be used as a translation of [0, 4, 18, 34] in 4D
+        without modification. An empty translation vector implies no
+        translation.
+    name : string
+        A string name for the transform.
+    """
+
+    def __init__(self, scale=(1.0,), translate=(0.0,), name=None):
+        super().__init__(name=name)
+        self.scale = np.array(scale)
+        self.translate = np.array(translate)
+
+    def __call__(self, coords):
+        coords = np.atleast_2d(coords)
+        scale = np.concatenate(
+            ([1.0] * (coords.shape[1] - len(self.scale)), self.scale)
+        )
+        translate = np.concatenate(
+            ([0.0] * (coords.shape[1] - len(self.translate)), self.translate)
+        )
+        return np.squeeze(scale * coords + translate)
+
+    @property
+    def inverse(self):
+        """Return the inverse transform."""
+        return ScaleTranslate(1 / self.scale, -1 / self.scale * self.translate)
+
+    def set_slice(self, axes: Sequence[int]):
+        """Return a transform subset to the visible dimensions."""
+        return ScaleTranslate(self.scale[axes], self.translate[axes])
+
+    def set_pad(self, axes: Sequence[int]):
+        """Return a transform with added axes for non-visible dimensions."""
+        n = len(axes) + len(self.scale)
+        not_axes = [i for i in range(n) if i not in axes]
+        scale = np.ones(n)
+        scale[not_axes] = self.scale
+        translate = np.zeros(n)
+        translate[not_axes] = self.translate
+        return ScaleTranslate(scale, translate)
+
+    def compose(self, transform):
```

Comment:
You should probably specify that this requires an STTransform. You can do this with strings if you don't want to require Py3.7+. I'll note that I am personally ok with dropping 3.6 early, because we are still alpha.

Reply:
I have gone with the string approach. Easy to switch over in the future if we want, but I'd rather wait to when we know we will use them (which could be very shortly). No need to make things unpleasant for our python 3.6 users until we need to.